pkgs/applications/emulators/emulationstation/default.nix

```diff
@@ -1,34 +1,72 @@
-{ lib, stdenv, fetchFromGitHub, pkg-config, cmake, curl, boost, eigen
-, freeimage, freetype, libGLU, libGL, SDL2, alsa-lib, libarchive
-, fetchpatch }:
+{ lib
+, SDL2
+, alsa-lib
+, boost
+, cmake
+, curl
+, eigen
+, fetchFromGitHub
+, fetchpatch
+, freeimage
+, freetype
+, libGL
+, libGLU
+, libarchive
+, pkg-config
+, stdenv
+, vlc
+, rapidjson
+, pugixml
+}:
 
 stdenv.mkDerivation {
   pname = "emulationstation";
-  version = "2.0.1a";
+  version = "2.12.0rp-dev";
 
   src = fetchFromGitHub {
-    owner = "Aloshi";
+    owner = "RetroPie";
     repo = "EmulationStation";
-    rev = "646bede3d9ec0acf0ae378415edac136774a66c5";
-    sha256 = "0cm0sq2wri2l9cvab1l0g02za59q7klj0h3p028vr96n6njj4w9v";
+    rev = "30cbdeb7cd667ac9d6a836995a17004e8d84339a";
+    sha256 = "sha256-jA3wLf/s9UlsP8h66aMOLZmLfTu7Zjhmt+BX6lmBreM=";
   };
 
-  patches = [
-    (fetchpatch {
-      url = "https://github.com/Aloshi/EmulationStation/commit/49ccd8fc7a7b1dfd974fc57eb13317c42842f22c.patch";
-      sha256 = "1v5d81l7bav0k5z4vybrc3rjcysph6lkm5pcfr6m42wlz7jmjw0p";
-    })
-  ];
+  patchPhase = ''
+    substituteInPlace */src/*.cpp */src/*/*.cpp */src/*.h \
+      --replace '<pugixml/src/pugixml.hpp>' '<pugixml.hpp>'
+    substituteInPlace external/CMakeLists.txt \
+      --replace 'add_subdirectory("pugixml")' 'find_package(pugixml REQUIRED)'
 
-  postPatch = ''
-    sed -i "7i #include <stack>" es-app/src/views/gamelist/ISimpleGameListView.h
+    substituteInPlace es-core/src/resources/ResourceManager.cpp \
+      --replace 'Utils::FileSystem::getHomePath() + "/.emulationstation/resources/"' "std::string{\"$out/resources/\"}"
```

Comment:
Does it really work? AFAICS it looks under user HOME dir, allowing each user to customize the resources.

Reply:
It works, yes. They are [checking multiple paths here](https://github.com/RetroPie/EmulationStation/blob/master/es-core/src/resources/ResourceManager.cpp#L31).
I can change it to one of the other tries, so it's easier to customize it.

I wanted to see first if this kind of patching seems acceptable to others at all.